indico/modules/categories/controllers/management.py

```diff
@@ -512,3 +514,40 @@ def _process(self):
                 self.role.members.add(user)
                 logger.info('User %r added to role %r by %r', user, self.role, session.user)
         return jsonify_data(html=_render_role(self.role, collapsed=False))
+
+
+class RHAddCategoryRoleMembersImportCSV(RHManageCategoryRole):
+    """Add users to a category role from CSV"""
+
+    def _process(self):
+        form = ImportMembersCSVForm()
+
+        if form.validate_on_submit():
+            (
+                csv_email,
+                registered_users,
+                registered_users_email,
+                not_registered_email,
+            ) = import_members_from_csv(form.source_file.data)
+            new_members = registered_users - self.role.members
+            not_registered = csv_email - registered_users_email
+            if bool(new_members):
+                if form.remove_existing.data:
+                    self.role.members = registered_users
+                    flash(ngettext("{} members has been imported.",
+                                   "{} members have been imported.",
+                                   registered_users.__len__()).format(registered_users.__len__()), 'success')
+                else:
+                    self.role.members |= new_members
+                    flash(ngettext("{} new member has been imported.",
+                                   "{} new members have been imported.",
+                                   new_members.__len__()).format(new_members.__len__()), 'success')
+            else:
+                flash(ngettext("{} member is already present.",
+                               "{} members are already present.",
+                               registered_users.__len__()).format(registered_users.__len__()), 'success')
+
+            if bool(not_registered):
+                flash(_("{} not registered email.".format(', '.join(not_registered))), 'warning')
```

Comment:
you need to do the `.format()` outside the `_()` call - the string in there must be fully static.

also, the text is strange.. what about something like "There are no users with these email addresses: ..." (with proper pluralization of course)